docs/Users/Tutorials/External_Bears.rst

```diff
@@ -0,0 +1,529 @@
+                    {"start", {
+                        {"column", column},
+                        {"file", in["filename"]},
+                        {"line", line}
+                    }},
+                    {"end", {
+                        {"column", column+6},
+                        {"file", in["filename"]},
+                        {"line", line}
+                    }}
+                }})},
+            {"severity", severity}
+        };
+        out["results"] += result;
+    }
+
+    int main() {
+
+        cin >> in;
+
+        init_results("coalaCheckBear");
+
+        cout << out;
+        return 0;
+    }
+
+.. note::
+
+    The ``C++`` operators and syntax are not well suited for JSON manipulation
+    but nlohmann's JSON lib makes it as easy as possible.
+
+12. Iterate over the lines and check for ``"coala"`` with an uppercase ``"C"``.
+    Use ``string``'s ``find`` function like so:
+
+::
+
+    #include <iostream>
+    #include <string>
+
+    #include "json.hpp"
+
+    using json = nlohmann::json;
+    using namespace std;
+
+    json in;
+    json out;
+    string origin;
+
+    void init_results(string bear_name) {
+        origin = bear_name;
+        out["results"] = json::array({});
+    }
+
+    void add_result(string message, int line, int column, int severity) {
+        json result = {
+            {"origin", origin},
+            {"message", message},
+                {"affected_code", json::array({{
+                    {"file", in["filename"]},
+                    {"start", {
+                        {"column", column},
+                        {"file", in["filename"]},
+                        {"line", line}
+                    }},
+                    {"end", {
+                        {"column", column+6},
+                        {"file", in["filename"]},
+                        {"line", line}
+                    }}
+                }})},
+            {"severity", severity}
+        };
+        out["results"] += result;
+    }
+
+    int main() {
+
+        cin >> in;
+
+        init_results("coalaCheckBear");
+
+        int i = 0;
+        for (auto it=in["file"].begin(); it !=in["file"].end(); it++) {
+            i++;
+            string line = *it;
+            size_t found = line.find("Coala");
+            while (found != string::npos) {
+                add_result("Did you mean 'coala'?", i, found, 2);
+                found = line.find("Coala", found+1);
+            }
+        }
+
+        cout << out;
+
+        return 0;
+    }
+
+13. After building the executable it has to be added it to the ``PATH`` env
+    variable. It is possible to modify the wrapper and give there the full path.
+    Add the current directory to the ``PATH`` like so:
+
+::
+
+    $ export PATH=$PATH:$PWD
+
+The last step is to test if everything is working properly. This is the testfile
+used in this tutorial (
+`testfile <https://github.com/Redridge/coalaCheckBear-cpp>`__).
```

Comment:
did you mean the link to be https://raw.githubusercontent.com/Redridge/coalaCheckBear-cpp/master/testfile?